Azure-Samples/chat-with-your-data-solution-accelerator
# Create reusable github workflow for docker build and push to reduce duplication.

## Purpose
We have three docker github workflows - one each for the frontend webapp, the backend (aka the frontend api) and the admin weapp.

* This PR extracts the common logic between these three workflows and parameterizes the registry, username and app names. 
* This PR also adds a workflow to run docker builds without push, that is configured to run on PR to main. This allows users to catch any issues with docker build before the code is merged to main.

Fixes #605. 

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x ] Other... Updating CI/CD config for docker image creation.
```

## How to Test
Check the passing builds on this PR - they should contain the "build without push" workflow jobs.
